src/EnhancedEntityRevisionTrait.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\src\EnhancedEntityRevisionTrait.
+ */
+
+namespace Drupal\src;
+
+/**
+ * Provides a trait implementing \Drupal\entity\EnhanceredEntityRevisionInterface.
+ */
+trait EnhancedEntityRevisionTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRevisionCreationTime() {
+    return $this->revision_create->value;
```

Comment:
Totally good questions.

a) Yeah I think we should have a hasField() method. For now I went with implement sane fallbacks, seems for me quite reasonable, like for example a revision log isn't critical, so skipping it, is kinda fine. Do you have strong opinions about exceptions?
b) Well yeah either we swap out the node class, or we don't support core classes in the first iteration